spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java

```diff
@@ -519,6 +526,20 @@ private BinaryExceptionClassifierBuilder classifierBuilder() {
 		return this.classifierBuilder;
 	}
 
+	/**
+	 * Configure to set DLT routing rules causing the message to be redirected to the custom
+	 * DLT when the configured exception has been thrown during message processing.
+	 * The cause of the originally thrown exception will be traversed in order to find the
+	 * match with the configured exceptions.
+	 * @param dltRoutingRules specification of custom DLT name extensions and exceptions which should be matched for them
+	 * @return the builder
+	 * @since 3.2.0
+	 */
+	public RetryTopicConfigurationBuilder dltRoutingRules(Map<String, Set<Class<? extends Throwable>>> dltRoutingRules) {
+		this.dltRoutingRules = dltRoutingRules;
```

Comment:
This is not OK. Has to be a copy.
The point is to avoid internal mutation when that external map is edited.
Well, that `dltRoutingRules` could be final and here you could just use `putAll()`